### ISSUE TITLE
Bad calculation in statsmodels.stats.weightstats.DescrStatsW

### ISSUE THREAD

**Issue by alandema** [statsmodels/statsmodels#8254]
#### Describe the bug

Trying to understand the calculations in DescrStatsW.tconfint_mean() I got confused with some method calls.

By the comments, why the number of observations should be equal to the to sum of weights?.

![image](https://user-images.githubusercontent.com/68517641/165620656-d9568e22-85e4-4671-ab16-cb11797227a1.png)

Also, in the method tconfint_mean, the dof is equal to (sum_weights - 1), but it should be equal to the number of observations -1.

![image](https://user-images.githubusercontent.com/68517641/165621613-37a771c4-436b-4412-8da9-dfcbc65df29e.png)

Also, in the std_mean(), the divisions use sum_weights, but it should use the number of observations.

![image](https://user-images.githubusercontent.com/68517641/165623829-8f81b4e5-043a-48fa-95f7-77f6558e124d.png)

**Comment by josef-pkt** [comment]
`weights` are frequency or case weights
So, the total number of observations is the sum of weights.

This was written before we disambiguated `weights`.
GLM has `freq_weights` and `var_weights` which are the unambiguous names.
WLS weights are `var_weights` for inference.
`weightstats` weights are `freq_weights`

eventually we need to rename this.

As a workaround
freq_weights can be normalized so that `freq_weights.sum() = nobs = len(x)`, by dividing weights by the desired nobs.

Aside: unit tests check weights by replicating each nobs according to it's (integer)weight, but weights don't have to be integers.

**Comment by alandema** [comment]
Indeed, in the case of frequencies the weights sums up to nobs and I was thinking in my case, where my weights are not frequencies and also are already normalized, so it always will sums up to 1.

By the docstring, I thought that using ddof = 0 was the same as to not use ddof and thus the sum of weights could be not equal to nobs, but this is false.

My suggestion for the docstring is to clarify that the weights MUST sums to the number of observation, in any case. But it would be better if the class was edited to work with freq_weights and var_weights.

**Comment by josef-pkt** [comment]
For var_weights you can also use WLS with a constant as the only regressor to get mean and mean inference. I did this to compute some regression equivalents to basic meta-analysis.

For var_weights in weightstats: 
I'm not sure we can get all statistics that are in methods defined for var_weights. I never looked at those in detail.
WLS, GLM and meta-analysis are easier because they produce results that are well defined for var_weights.
(WLS also has an option to fix the scale. residual variance at 1 if the var_weights are assumed to reflect all uncertainty.)

e.g. https://www.statsmodels.org/dev/examples/notebooks/generated/metaanalysis1.html

**Comment by alandema** [comment]
I will have a look into WLS, thanks.
